homeassistant/components/generic/config_flow.py

```diff
@@ -261,6 +265,17 @@ async def async_test_stream(
     return {}
 
 
+def register_preview(hass: HomeAssistant, flow_id: str, preview: GenericCamera):
+    """Set up previews for camera feeds during config flow."""
+    hass.data.setdefault(DOMAIN, {})
+
+    if not hass.data[DOMAIN].get(PREVIEWS):
+        _LOGGER.debug("Registering camera image preview handler")
+        hass.http.register_view(CameraImagePreview(hass))
+        hass.data[DOMAIN][PREVIEWS] = {}
+    hass.data[DOMAIN][PREVIEWS][flow_id] = preview
```

Comment:
I'm also realizing there may be considerations about the entity life cycle by reusing the entity. For example, we may create a stream object which need to be stopped when this object is done. Do you have thoughts about that?

Reply:
Yes, I was thinking about this.  I assume the garbage collector means that instances are preserved until all references are gone.  Given that there isn't (or at least I couldn't find) something that cancels the flow if the user times out/ignores it, there's a risk that we wouldn't even be able to trigger the code that removed the reference.

In that case it might be simpler/safer to do as you suggest and dynamically generate the camera view in the http response, pull an image from it then remove it (perhaps explicitly) before returning the image to the user.

In all of these cases it would be really handy to have an API to get a reference to the actual running config flow object rather than have to copy data/references etc.  I think the API addition would be very small.

Thinking slightly further ahead I'm hoping to do a follow up PR that can preview the stream as well (not sure about the details of it yet!).

I'll start with the improvements you've suggested - thanks for the review!